frontend/app/utils/string-utils.ts

```diff
@@ -7,3 +7,12 @@
 export function expandTemplate(template: string, variables: Record<string, string>) {
   return Object.entries(variables).reduce((t, [key, value]) => t.replace(`{${key}}`, value), template);
 }
+
+/**
+ * Generate a random string using only hex characters.
+ */
+export function randomHexString(len: number) {
+  const allowedChars = '0123456789abcdefghijklmnopqrstuvwxyz';
```

Comment:
shouldn't this be `const allowedChars = '0123456789abcdef'`?

Reply:
Fixed. Please re-review.